projects/batfish-common-protocol/src/main/java/org/batfish/storage/FileStorageImpl.java

```diff
@@ -0,0 +1,158 @@
+package org.batfish.storage;
+
+import com.google.common.collect.Sets;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.batfish.common.BatfishException;
+import org.batfish.common.BfConsts;
+import org.batfish.common.util.CommonUtil;
+import org.batfish.datamodel.pojo.Analysis;
+
+public class FileStorageImpl implements Storage {
+
+  private final Path _containersLocation;
+
+  public FileStorageImpl(Path containersLocation) throws BatfishException {
+    try {
+      if (containersLocation != null) {
+        _containersLocation = containersLocation;
+        _containersLocation.toFile().mkdir();
+      } else {
+        throw new BatfishException("container location is null");
+      }
+    } catch (InvalidPathException e) {
+      throw new BatfishException("cannot resolve containers location '" + containersLocation + "'");
+    }
+  }
+
+  /**
+   * Retrieve a container Path
+   *
+   * @param containerName Container name
+   * @return Path of the container
+   */
+  // deprecated
+  @Override
+  public Path getContainerPath(String containerName) {
```

Comment:
Should be a private method?

Reply:
This method was to be used to get container path using container name from storage apis for the places where container path was necessary. But I think this is not being used anywhere right now and is irrelevant with respect to this pull request, since we are only targeting analysis; I will remove this function

Reply:
I thought we shouldn't provide any public functions that returns any file system path~ Maybe I was wrong~ :)

Reply:
No, you are correct, it was for backward compatibility but right now noone is using it so it can be removed

Reply:
Done